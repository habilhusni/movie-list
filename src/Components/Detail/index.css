.card-detail {
  width: 500px !important;
  margin: auto !important;
}

.img-poster {
  width: 100px;
}
