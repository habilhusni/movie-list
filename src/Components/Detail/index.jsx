import React from "react";
import { Descriptions } from "antd";
import "./index.css";

const isDetailsAvailable = (dtls) => {
  if (!dtls) {
    return false;
  } else if (!dtls.payload) {
    return false;
  } else if (!dtls.payload.data) {
    return false;
  }

  return true;
};

const Detail = (props) =>
  isDetailsAvailable(props.details) ? (
    <Descriptions title={props.details.payload.data.Title} bordered>
      <Descriptions.Item label="Actors">
        {props.details.payload.data.Actors}
      </Descriptions.Item>
      <Descriptions.Item label="Awards">
        {props.details.payload.data.Awards}
      </Descriptions.Item>
      <Descriptions.Item label="Box Office">
        {props.details.payload.data.BoxOffice}
      </Descriptions.Item>
      <Descriptions.Item label="Country">
        {props.details.payload.data.Country}
      </Descriptions.Item>
      <Descriptions.Item label="DVD">
        {props.details.payload.data.DVD}
      </Descriptions.Item>
      <Descriptions.Item label="Director">
        {props.details.payload.data.Director}
      </Descriptions.Item>
      <Descriptions.Item label="Genre">
        {props.details.payload.data.Genre}
      </Descriptions.Item>
      <Descriptions.Item label="Language">
        {props.details.payload.data.Language}
      </Descriptions.Item>
      <Descriptions.Item label="Metascore">
        {props.details.payload.data.Metascore}
      </Descriptions.Item>
      <Descriptions.Item label="Plot">
        {props.details.payload.data.Plot}
      </Descriptions.Item>
      <Descriptions.Item label="Poster">
        <img
          src={props.details.payload.data.Poster}
          alt="poster"
          className="img-poster"
        />
      </Descriptions.Item>
      <Descriptions.Item label="Production">
        {props.details.payload.data.Production}
      </Descriptions.Item>
      <Descriptions.Item label="Rated">
        {props.details.payload.data.Rated}
      </Descriptions.Item>
      <Descriptions.Item label="Released">
        {props.details.payload.data.Released}
      </Descriptions.Item>
      <Descriptions.Item label="Response">
        {props.details.payload.data.Response}
      </Descriptions.Item>
      <Descriptions.Item label="Runtime">
        {props.details.payload.data.Runtime}
      </Descriptions.Item>
      <Descriptions.Item label="Type">
        {props.details.payload.data.Type}
      </Descriptions.Item>
      <Descriptions.Item label="Website">
        {props.details.payload.data.Website}
      </Descriptions.Item>
      <Descriptions.Item label="Writer">
        {props.details.payload.data.Writer}
      </Descriptions.Item>
      <Descriptions.Item label="Year">
        {props.details.payload.data.Year}
      </Descriptions.Item>
      <Descriptions.Item label="imdbID">
        {props.details.payload.data.imdbID}
      </Descriptions.Item>
      <Descriptions.Item label="imdbRating">
        {props.details.payload.data.imdbRating}
      </Descriptions.Item>
      <Descriptions.Item label="imdbVotes">
        {props.details.payload.data.imdbVotes}
      </Descriptions.Item>
    </Descriptions>
  ) : (
    <h2>No details</h2>
  );

export default Detail;
