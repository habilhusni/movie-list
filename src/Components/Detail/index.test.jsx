import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";

import Detail from "./index.jsx";

Enzyme.configure({ adapter: new Adapter() });

describe("Test Snapshot and Functional", () => {
  let component;
  let props = {
    details: {
      payload: {
        data: {
          Title: "Kucing",
          Actors: "Actors",
          Awards: "Awards",
          BoxOffice: "BoxOffice",
          Country: "Country",
          DVD: "DVD",
          Director: "Director",
          Genre: "Genre",
          Language: "Language",
          Metascore: "Metascore",
          Plot: "Plot",
          Poster: "Poster",
          Production: "Production",
          Rated: "Rated",
          Released: "Released",
          Response: "Response",
          Runtime: "Runtime",
          Type: "Type",
          Website: "Website",
          Writer: "Writer",
          Year: "Year",
          imdbID: "imdbID",
          imdbRating: "imdbRating",
          imdbVotes: "imdbVotes",
        },
      },
    },
  };

  beforeEach(() => {
    component = shallow(<Detail {...props} />);
  });

  it("Testing Detail Snapshot", () => {
    expect(toJSON(component)).toMatchSnapshot();
  });
});
