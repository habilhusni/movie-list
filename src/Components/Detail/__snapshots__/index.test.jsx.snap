// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Snapshot and Functional Testing Detail Snapshot 1`] = `
<Descriptions
  bordered={true}
  title="Kucing"
>
  <DescriptionsItem
    label="Actors"
  >
    Actors
  </DescriptionsItem>
  <DescriptionsItem
    label="Awards"
  >
    Awards
  </DescriptionsItem>
  <DescriptionsItem
    label="Box Office"
  >
    BoxOffice
  </DescriptionsItem>
  <DescriptionsItem
    label="Country"
  >
    Country
  </DescriptionsItem>
  <DescriptionsItem
    label="DVD"
  >
    DVD
  </DescriptionsItem>
  <DescriptionsItem
    label="Director"
  >
    Director
  </DescriptionsItem>
  <DescriptionsItem
    label="Genre"
  >
    Genre
  </DescriptionsItem>
  <DescriptionsItem
    label="Language"
  >
    Language
  </DescriptionsItem>
  <DescriptionsItem
    label="Metascore"
  >
    Metascore
  </DescriptionsItem>
  <DescriptionsItem
    label="Plot"
  >
    Plot
  </DescriptionsItem>
  <DescriptionsItem
    label="Poster"
  >
    <img
      alt="poster"
      className="img-poster"
      src="Poster"
    />
  </DescriptionsItem>
  <DescriptionsItem
    label="Production"
  >
    Production
  </DescriptionsItem>
  <DescriptionsItem
    label="Rated"
  >
    Rated
  </DescriptionsItem>
  <DescriptionsItem
    label="Released"
  >
    Released
  </DescriptionsItem>
  <DescriptionsItem
    label="Response"
  >
    Response
  </DescriptionsItem>
  <DescriptionsItem
    label="Runtime"
  >
    Runtime
  </DescriptionsItem>
  <DescriptionsItem
    label="Type"
  >
    Type
  </DescriptionsItem>
  <DescriptionsItem
    label="Website"
  >
    Website
  </DescriptionsItem>
  <DescriptionsItem
    label="Writer"
  >
    Writer
  </DescriptionsItem>
  <DescriptionsItem
    label="Year"
  >
    Year
  </DescriptionsItem>
  <DescriptionsItem
    label="imdbID"
  >
    imdbID
  </DescriptionsItem>
  <DescriptionsItem
    label="imdbRating"
  >
    imdbRating
  </DescriptionsItem>
  <DescriptionsItem
    label="imdbVotes"
  >
    imdbVotes
  </DescriptionsItem>
</Descriptions>
`;
